packages/pwa-kit-react-sdk/src/ssr/universal/components/refresh/index.jsx

```diff
@@ -39,6 +39,19 @@ const Refresh = () => {
                 console.warn('Could not find `referrer` search param - redirecting to home page.')
                 referrer = '/'
             }
+
+            // if experimentalUnsafeReloadServerSide is true, reload the page on server side
+            const experimentalUnsafeReloadServerSide = new URLSearchParams(location.search).get(
+                'experimentalUnsafeReloadServerSide'
+            )
+            if (experimentalUnsafeReloadServerSide === 'true') {
+                console.warn(
+                    '"experimentalUnsafeReloadServerSide" was set to true - reloading page on server side'
+                )
+                window.location.assign(referrer)
```

Comment:
To match the logic on the Runtime Admin's side (specifically in the preview client script), we'll need to make sure that the `location` history remains intact and not added with a new entry. So that's why `replace` is needed here.

```suggestion
                window.location.replace(referrer)
```

Reply:
all set